tests/oauth2/test_oauth2_admin.py

```diff
@@ -0,0 +1,109 @@
+from collections.abc import Set
+from unittest.mock import patch
+from regtech_api_commons.oauth2.oauth2_admin import OAuth2Admin
+import jose.jwt
+
+from conftest import kc_settings
+
+oauth2_admin = OAuth2Admin(kc_settings)
+
+
+def test_get_claims():
+    token = "Test Token"
+
+    with patch.object(oauth2_admin, "_get_keys", return_value="secret"):
+        claims = {
+            "token": token,
+            "iss": kc_settings.kc_realm_url.unicode_string(),
+            "audience": kc_settings.auth_client,
+            "options": kc_settings.jwt_opts,
+        }
+
+        encoded = jose.jwt.encode(claims=claims, key=oauth2_admin._get_keys(), algorithm="HS256")
+
+        expected_result = jose.jwt.decode(
+            token=encoded,
+            key=oauth2_admin._get_keys(),
+            issuer=kc_settings.kc_realm_url.unicode_string(),
+            audience=kc_settings.auth_client,
+            options=kc_settings.jwt_opts,
+        )
+
+        actual_result = oauth2_admin.get_claims(encoded)
+        assert actual_result == expected_result
+
+
+def test_update_user(mocker):
+
+    mock_update_user = mocker.patch("regtech_api_commons.oauth2.oauth2_admin.OAuth2Admin.update_user")
+    mock_update_user.return_value = None
+
+    res = oauth2_admin.update_user(user_id="test-user-id", payload={"Name": "Test"})
+    assert res is None
+
+    mock_get_user = mocker.patch("keycloak.KeycloakAdmin.get_user")
+    mock_get_user.return_value = {"Name": "Test"}
+
+    user = oauth2_admin._admin.get_user(user_id="test-user-id")
+    assert user["Name"] == "Test"
+
+
+def test_upsert_group(mocker):
+    lei = "TESTLEI"
+    name = "Test Name"
+
+    mock_get_group = mocker.patch("regtech_api_commons.oauth2.oauth2_admin.OAuth2Admin.get_group")
+    mock_get_group.return_value = {"id": lei, "Name": name}
+
+    mock_update_group = mocker.patch("keycloak.KeycloakAdmin.update_group")
+    mock_update_group.return_value = lei
+
+    result = oauth2_admin.upsert_group(lei=lei, name=name)
+    assert result == lei
+
+
+def test_get_group(mocker):
+    lei = "TESTLEI"
+    name = "Test Name"
+
+    mock_get_group = mocker.patch("regtech_api_commons.oauth2.oauth2_admin.OAuth2Admin.get_group")
+    mock_get_group.return_value = {"id": lei, "Name": name}
```

Comment:
I think the goal here would be to patch the KeycloakAdmin get_group_by_path function, similar to how you patched the update_group during the upsert.  Othewise, you're bypassing our "get_group" function, which is what we're really trying to test.